Initial commit of SDL2 (2.0.4)

jimklimov opened the pull request:
Added a recipe for libsdl2 and friends, since sdl1.2.x is more or less considered obsolete by the project, and I'm looking at some programs that would need SDL2 to compile.
Package builds and publishes cleanly. Usability for the job - remains to be seen ;)

jimklimov committed:
Initial commit of SDL2 (2.0.4)

pyhalov commented on components/SDL2/Makefile:
```diff
@@ -0,0 +1,56 @@
+
+CONFIGURE_OPTIONS +=	--sysconfdir=/etc
+CONFIGURE_OPTIONS +=	--libexecdir=$(CONFIGURE_LIBDIR.$(BITS))
+CONFIGURE_OPTIONS +=	--libdir=$(CONFIGURE_LIBDIR.$(BITS))
```
This should be automatic

jimklimov replied:
Copypasted from SDL component...

pyhalov commented on components/SDL2/Makefile:
```diff
@@ -0,0 +1,56 @@
+
+COMPONENT_NAME=		SDL2
+COMPONENT_VERSION=	2.0.4
+COMPONENT_REVISION=	1
```
COMPONENT_REVISION is not necessary for the first revision

pyhalov commented on components/SDL2/sdl2.p5m:
```diff
@@ -0,0 +1,127 @@
+file path=usr/lib/libSDL2.a
+file path=usr/lib/libSDL2_test.a
+file path=usr/lib/libSDL2main.a
+dir  path=usr/lib/cmake
```
We usually avoid delivering dir actions until it's  strictly necessary

jimklimov replied:
and do we commonly have /usr/lib/cmake in a typical installation? which might not have cmake itself?

pyhalov replied:
It doesn't matter, file action for usr/lib/cmake/SDL2/sdl2-config.cmake is enough

jimklimov replied:
fixed

pyhalov commented on components/SDL2/Makefile:
```diff
@@ -0,0 +1,55 @@
+			( cd $(@D) && $(CONFIG_SHELL) autogen.sh )
+
+#CONFIGURE_OPTIONS +=	--sysconfdir=/etc
+#CONFIGURE_OPTIONS +=	--libexecdir=$(CONFIGURE_LIBDIR.$(BITS))
```
--libexecdir and --sysconfdir are not passed by default, but --libdir is added by default rules (see configure.mk)

jimklimov commented:
Hopefully addressed all the comments above (rebuilding and committing soon).

On a side note, do we have a good mechanism to use different HW-optimized features or builds as determined at runtime (like loopback mounts of libc.so back in the day)? (Anyhow a task for some other non-initial development and PR)

I ask because the SDL2 has a number of compile-time options for different CPU instructions that might make a difference on newer CPUs but might not work on older ones, but if I understand it correctly - it does not offer building a single binary that would pick correct implementation based on CPU capabilities, sadly...

pyhalov commented:
No, we don't have such mechanism. 
Please, move files to components/library/SDL2

jimklimov commented:
Is this good to go? IIRC it was a dependency for OpenTTD to be built... ;)

pyhalov commented:
Yes, please, merge commits.
